### PR TITLE
created the api router endpoints for orders #31

### PR DESCRIPTION
**Pull request by DonnyDunamis**
These are the api routing endpoints for orders ready for review.

### CODE REVIEW

File: api/orders.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+} = require("../db");
+const { requireUser } = require("./utils");
+
+router.get("/", async (req, res, next) => {
+  const orders = await getAllOrders();
+
+  res.send(orders);
+});
+
+router.post("/", requireUser, async (req, res) => {
```
Comment (scburkha005) [inline api/orders.js]: we don't need to requireUser - guest accounts should make orders too

File: api/orders.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    orderData.subtotal = subtotal;
+
+    const order = await createOrder(orderData);
+
+    if (!order) {
+      next({
+        name: "UserNotAuthorized",
+        message: "User is not authorized to create an order.",
+      });
+    }
```
Comment (scburkha005) [inline api/orders.js:33]: change error message: something went wrong creating your order? im not sure what a good message here is

File: api/orders.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        id: orderId,
+        subtotal,
+      });
+      res.send(updatedOrder);
+    } else {
+      next({
+        name: "userUnauthorizeToUpdate",
+        message: "User is not authorized to update this order.",
+      });
+    }
```
Comment (scburkha005) [inline api/orders.js:59]: if (!orderById.creatorId) {
// guest order
} else if (orderById.creatorId === req.user.id) {
// user order
}

File: api/orders.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    const orderById = await getOrderById(orderId);
+    if (orderById.creatorId === req.user.id) {
+      const destroyedOrder = await destroyOrder(orderId);
+      res.send(destroyedOrder);
+    } else {
+      next({
+        name: "UserUnauthorizeToDelete",
+        message: "User is not authorized to delete this order.",
+      });
+    }
```
Comment (scburkha005) [inline api/orders.js:77]: reference above comment if else if structure

File: api/orders.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    next({ name, message });
+  }
+});
+
+router.delete("/:orderId", async (req, res, next) => {
+  const { orderId } = req.params;
+  try {
+    const orderById = await getOrderById(orderId);
+    if (orderById.creatorId === req.user.id) {
+      const destroyedOrder = await destroyOrder(orderId);
```
Comment (scburkha005) [inline api/orders.js:70]: destroyedOrderId rename?

File: package.json [hunk 6]
```diff
@@ -30,7 +30,8 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-router-dom": "^6.2.2",
-    "react-scripts": "^5.0.0"
+    "react-scripts": "^5.0.0",
+    "utils": "^0.3.1"
```
Comment (scburkha005) [inline package.json:34]: delete this line
npm uninstall utils

**Review by scburkha005 — APPROVED** [review]
minor changes req

File: api/orders.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  addProductToOrder,
+  destroyOrder,
+} = require("../db");
+const { requireUser } = require("./utils");
+
+router.get("/", async (req, res, next) => {
+  const orders = await getAllOrders();
+
+  res.send(orders);
+});
```
[the remark this answers is not in the document]
Comment (DonnyDunamis) [reply api/orders.js]: Done.

File: api/orders.js (hunk 1 above)
Comment (DonnyDunamis) [reply api/orders.js]: Done.

File: api/orders.js (hunk 2 above)
Comment (DonnyDunamis) [reply api/orders.js]: I changed it. Does that work?

File: api/orders.js (hunk 5 above)
Comment (DonnyDunamis) [reply api/orders.js]: Done.

File: package.json (hunk 6 above)
Comment (DonnyDunamis) [reply package.json]: Done.

File: api/orders.js (hunk 3 above)
Comment (DonnyDunamis) [reply api/orders.js]: Done, but check it before we resolve the conversation.

File: api/orders.js (hunk 4 above)
Comment (DonnyDunamis) [reply api/orders.js]: Done, but please review the code, like with the patch route.

**Review by scburkha005 — APPROVED** [review]
nice work man 👍

**Pull request merged**